pkg/controllers/report/utils/scanner.go

```diff
@@ -105,3 +116,29 @@ func (s *scanner) validateImages(ctx context.Context, resource unstructured.Unst
 	}
 	return &response, nil
 }
+
+func(s *scanner) ScanResourceForVAPs(ctx context.Context, resource unstructured.Unstructured, nsLabels map[string]string, policies ...v1alpha1.ValidatingAdmissionPolicy) map[v1alpha1.ValidatingAdmissionPolicy]ScanResult {
+	results := map[v1alpha1.ValidatingAdmissionPolicy]ScanResult{}
+	for _, policy := range policies {
+		var errors []error
+		logger := s.logger.WithValues("kind", resource.GetKind(), "namespace", resource.GetNamespace(), "name", resource.GetName())
+		applyPolicy := common.ApplyPolicyConfig{
+			ValidatingAdmissionPolicy : policy,
+		}
+		policyDecisions, err := common.ApplyPolicyOnResource(applyPolicy)
+		if err != nil {
+			logger.Error(err, "failed to scan resource")
+			errors = append(errors, err)
+		}
+		for _, decision := range policyDecisions {
+			if strings.Compare(string(decision.Action), "deny") == 0 {
```

Comment:
It won't work. 
First you need to create a new variable of type `common.ValidatingAdmissionPolicies{}` then loop for all vap policies. For each vap policy, create an `ApplyPolicyConfig` that takes both policy and resource, then you can call `ApplyPolicyOnResource`

```
validatingAdmissionPolicy := common.ValidatingAdmissionPolicies{}
for _, policy := range policies {
               applyConfig := common.ApplyPolicyConfig{
			ValidatingAdmissionPolicy : policy,
                        resource: resource,
		}
		engineResp, err := validatingAdmissionPolicy .ApplyPolicyOnResource(applyConfig)

                // Construct the result here
}
```

Reply:
Yes, I have added the resource locally in the current update. I have a doubt about how to compare the Action as the decision is of engineapi.EngineResponse type and we require validatingadmissionpolicy.PolicyDecision type as the PolicyDecision struct contains the field Action and Message.

Reply:
I believe it works even if we don't create a new variable of type common.ValidatingAdmissionPolicies{}.